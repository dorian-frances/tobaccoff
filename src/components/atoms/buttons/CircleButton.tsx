import { TouchableRipple } from 'react-native-paper';
import Icon from '@expo/vector-icons/MaterialCommunityIcons';
import React from 'react';
import { View } from 'react-native';
import { ColorsEnum } from '../../../assets/colors/colors.enum';

type CircleButtonProps = {
  testID?: string;
};

const CircleButton = ({ testID }: CircleButtonProps) => {
  return (
    <TouchableRipple>
      <View
        style={{
          height: 70,
          width: 70,
          borderRadius: 100,
          backgroundColor: ColorsEnum.INPUT_BACKGROUND_COLOR,
          elevation: 2,
          justifyContent: 'center',
          alignItems: 'center',
        }}
        testID={testID}
      >
        <Icon
          name={'restart'}
          size={40}
          color={ColorsEnum.INDICATIVE_TEXT_COLOR}
        />
      </View>
    </TouchableRipple>
  );
};

export default CircleButton;
