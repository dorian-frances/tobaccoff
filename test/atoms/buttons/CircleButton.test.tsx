import React from 'react';
import CircleButton from '../../../src/components/atoms/buttons/CircleButton';
import { render } from '@testing-library/react-native';

describe('<CircleButton />', () => {
  it('should render correctly', () => {
    const circleButton = render(<CircleButton />);
    expect(circleButton.toJSON).toMatchSnapshot();
  });

  it('should find the CircleButton by testId', () => {
    const circleButtonTestId = 'circleButtonTestId';
    const circleButton = render(<CircleButton testID={circleButtonTestId} />);
    const foundButtonView = circleButton.getByTestId(circleButtonTestId);
    expect(foundButtonView).toBeTruthy();
  });
});
