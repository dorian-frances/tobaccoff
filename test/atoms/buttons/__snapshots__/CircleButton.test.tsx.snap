// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleButton /> should render correctly 1`] = `[Function]`;
